ui/app/services/auth.js

```diff
@@ -446,6 +455,10 @@ export default Service.extend({
       return this._parseMfaResponse(resp.auth?.mfa_requirement);
     }
 
+    // this is the problem -- the auth/userpass/login/ endpoint returns a diff response than self-lookup. it doesn't include anything about expiration for batch tokens
+
+    console.log('authenticate resp: ', resp);
+
     return this.authSuccess(options, resp.auth || resp.data);
```

Comment:
i believe this is the root of the problem. depending on what login method the user is using (token, userpass, LDAP etc), we hit a different API endpoint, and those apis have a different response shape. note the difference between

<table>
<tr><th></th><th>endpoint</th><th>response</th></tr>
<tr>
<th>/auth/token/lookup-self</th>
<td>

<img width="831" alt="Screenshot 2024-06-12 at 7 16 15 PM" src="https://github.com/hashicorp/vault/assets/903288/8747255a-6a5d-42bf-87ce-f9d6fbb64d63">

</td>

<td>

<img width="1376" alt="Screenshot 2024-06-12 at 7 16 27 PM" src="https://github.com/hashicorp/vault/assets/903288/0c02654f-e558-423f-9294-d9f5a4e802b7">

</td>
</tr>

<tr>
<th>example 2</th>
<td>

<img width="1698" alt="Screenshot 2024-06-12 at 7 14 07 PM" src="https://github.com/hashicorp/vault/assets/903288/96541e05-a3bf-44ff-8dc8-aa8808ec902c">


</td>
<td>

<img width="1719" alt="Screenshot 2024-06-12 at 7 14 16 PM" src="https://github.com/hashicorp/vault/assets/903288/7aa27cbe-8fed-499f-9684-7d05b353a12d">


</td>
</tr>
</table>